scenes/lessons/7th_class/a_b+c+d/a_b+c+d.py

```diff
@@ -0,0 +1,616 @@
+from manim import MathTex, VGroup
+from manim import SurroundingRectangle, Rectangle, Circle
+from manim import RIGHT, LEFT, UP, DOWN, DR, UL, DR, ORIGIN
+from manim import AnimationGroup, Write, Unwrite, FadeIn, FadeOut, Create, ReplacementTransform
+from manim import Indicate, Circumscribe, Wiggle, TransformMatchingTex, ClockwiseTransform
+
+import numpy as np
+from colour import Color
+
+from aramanim import ConectionLine
+from hanrahashiv import FormulaModificationsScene, ModifyFormula
+
+BIG_SCALE = 1.3
+SCALE = .45
+
+
+def get_tex(expression, color=WHITE):
+    """
+    Receives a math expression, and attempts to parse that into a Tex object.
+    It's pretty basic and the main advantage is the readability of the expression.
+    Avoids the dollar signs and other ugly details.
+    WARNING: Don't use this for very complex formulas.
+    """
+    tex_list = list()
+    if '$\cdot$' in expression:
+        raise Exception("Please provide the dot multiply sign as '*'")
+    skip_char = False
+    for i, char in enumerate(expression):
+        if skip_char:
+            skip_char = False
+            continue
+        if char.isalpha():
+            tex_list.append(char)
+        elif char == '*':
+            tex_list.append('\cdot')
+        elif char == '^':
+            tex_list.append(f'^{expression[i + 1]}')
+            skip_char = True
+        elif char == ' ':
+            tex_list.append(' ')
+        else:
+            tex_list.append(char)
+
+    return MathTex(*tex_list, color=color, arg_separator=' ')
+
+
+def find_index(arr, val):
+    """
+    Given a string or an array [arr], find the index of the given char or value [val].
+    Ex: find_index('a*(a+b)', 'a') -> [0, 3]
+    """
+    if isinstance(arr, str):
+        arr = np.array([el for el in arr])
+    return np.where(arr == val)[0].tolist()
+
+
+def circle(mobject, color=WHITE, stroke_width=4.5, buff=0.08):
+    """
+    Defines a surrounding circle for the given mobject
+    """
+    c = Circle(color=color, stroke_width=stroke_width).surround(mobject, buffer_factor=1)
+    radius = c.width / 2
+    c.scale((radius + buff) / radius)
+    return c
+
+
+class scene1(FormulaModificationsScene):
+    def construct(self):
+        self.wait()
+
+        abc = get_tex('a b c').scale(BIG_SCALE)
+        equal_sign_1 = get_tex('=', color=YELLOW)
+        plus_sign_1, plus_sign_2, plus_sign_3, plus_sign_4 = (get_tex('+', color=YELLOW) for _ in range(4))
+
+        self.form_1_str = 'a*(b+c      )=a*b+a*c            '
+        self.form_2_str = 'a*(b+c+d    )=a*b+a*c+a*d        '
+        self.form_3_str = 'a*(b+c+d+e+f)=a*b+a*c+a*d+a*e+a*f'
+        a_idx, b_idx, c_idx, d_idx, e_idx, f_idx = [find_index(self.form_3_str, char) for char in 'abcdef']
+
+        form_1, form_2, form_3 = [
+            get_tex(form).scale(BIG_SCALE) 
+            for form in [self.form_1_str, self.form_2_str, self.form_3_str]
+        ]
+
+        [form_1[i].set_color(ORANGE) for i in a_idx]
```

Comment:
Aren't those form_3 indices?

Reply:
Because of these spaces, the indices are stored for all by using the form_3 string.

```
self.form_1_str = 'a*(b+c      )=a*b+a*c            '
self.form_2_str = 'a*(b+c+d    )=a*b+a*c+a*d        '
self.form_3_str = 'a*(b+c+d+e+f)=a*b+a*c+a*d+a*e+a*f'
```

Reply:
:O աստվածային